dev/Addi-11/kfold.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+from sklearn.model_selection import cross_val_score, cross_validate
+import matplotlib.pyplot as plt
+import pandas as pd
+from dataloader import get_x_y
+
+def performance_over_folds(clf, num_fold=10):
+    """
+	Evaluates the performance of the model over different fold sizes.
+
+	Parameters:
+		clf : trained classification model
+		num_fold : int
+
+	Returns:
+		fit_times: array-like
+		train_scores: array-like
+		test_scores: array-like
+	"""
+
+    folds = np.arange(2, num_fold + 1)
+    fit_times = np.zeros(len(folds))
+    train_scores = np.zeros(len(folds))
+    test_scores = np.zeros(len(folds))
+
+    x, y = get_x_y()
+
+    for i, fold in enumerate(folds):
+        fit_times[i] = np.mean(
+            cross_validate(clf, x, y, cv=fold, return_train_score=True)["fit_time"]
+        )
+        train_scores[i] = np.mean(
+            cross_validate(clf, x, y, cv=fold, return_train_score=True)["train_score"]
+        )
+        test_scores[i] = np.mean(
+            cross_validate(clf, x, y, cv=fold, return_train_score=True)["test_score"]
```

Comment:
It looks like you are rerunning the cross-validation 3 times?

Reply:
I apologise for the silly mistake.  Yes, I was suppose to run it once. 
Will work on it right away and also the changes suggested on the graphs.